kubevirt-vmware/pkg/controller/ovirtprovider/ovirtprovider_controller.go

```diff
@@ -0,0 +1,279 @@
+	// Create a new controller
+	c, err := controller.New("ovirtprovider-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource OVirtProvider
+	err = c.Watch(&source.Kind{Type: &v2vv1alpha1.OVirtProvider{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileOVirtProvider{}
+
+// ReconcileOVirtProvider reconciles a OVirtProvider object
+type ReconcileOVirtProvider struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a OVirtProvider object and makes changes based on the state read
+// and what is in the OVirtProvider.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileOVirtProvider) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling OVirtProvider")
+
+	// Fetch the OVirtProvider instance
+	instance := &v2vv1alpha1.OVirtProvider{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			reqLogger.Info("The request object cannot be found.")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		reqLogger.Info("Error reading the request object, requeuing.")
+		return reconcile.Result{}, err
+	}
+
+	connectionSecret, err := r.fetchSecret(instance)
+	if err != nil {
+		reqLogger.Error(err, "Failed to get Secret object for the oVirt connection")
+		return reconcile.Result{}, err // request will be re-queued
+	}
+	reqLogger.Info("Connection secret retrieved.")
+
+	if len(instance.Spec.Vms) == 0 { // list of oVirt VMs is requested to be retrieved
+		err = r.readVmsList(request, connectionSecret)
+		if err != nil {
+			reqLogger.Error(err, "Failed to read list of oVirt VMs.")
+			// TODO: check the reason to fail and decide whether to wait like for vmware or to fail the import
+		}
+
+		return reconcile.Result{}, err // request will be re-queued if failed
+	}
+
+	// secret is present, list of VMs is available, let's check for  details to be retrieved
+	var lastError error = nil
+	for _, vm := range instance.Spec.Vms { // sequential read is probably good enough, just a single VM or a few of them are expected to be retrieved this way
+		if vm.DetailRequest {
```

Comment:
I might be reading this wrong, but ` vm.DetailRequest` was set to false in line 96 `r.readVmsList(request, connectionSecret)` - which calls to https://github.com/ManageIQ/manageiq-v2v-conversion_host/pull/32/files#diff-d8622e4bd9da988af1519fe20faeb812R174 

this condition `if vm.DetailRequest` is expected to be false, isn't it ?

Reply:
UI can set it to `true` which means that controller need to fetch the details.

Reply:
Okay, so that will happen across two different loops - the UI presents the list of VMs and then pick which VM details should be update by setting `DetailRequest` to true ?

Reply:
Correct